src/browser/datauri.zig

```diff
@@ -0,0 +1,79 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+
+// Represents https://developer.mozilla.org/en-US/docs/Web/URI/Reference/Schemes/data
+pub const DataURI = struct {
+    was_base64_encoded: bool,
+    // The contents in the uri. It will be base64 decoded but not prepared in
+    // any way for mime.charset.
+    data: []const u8,
+
+    // Parses data:[<media-type>][;base64],<data>
+    pub fn parse(allocator: Allocator, src: []const u8) !DataURI {
+        if (!std.mem.startsWith(u8, src, "data:")) {
+            return error.Invalid;
+        }
+
+        const uri = src[5..];
+        const data_starts = std.mem.indexOf(u8, uri, ",") orelse return error.Invalid;
```

Comment:
Can use `std.mem.indexOfScalar` directly.  `std.mem.indexOf` _will_ call it, but only after doing 2 length checks on the needle.